spec/parser/ruby/legacy/token_list_spec.rb

```diff
@@ -1,18 +1,23 @@
 # frozen_string_literal: true
 
-include YARD::Parser::Ruby::Legacy
-include YARD::Parser::Ruby::Legacy::RubyToken
+# Note: including YARD::Parser::Ruby::Legacy at top-level is problematic
```

Comment:
You don't need this note, I think your solution is the right approach regardless of the reason to move this.